source/common/runtime/runtime_features.cc

```diff
@@ -102,6 +102,8 @@ constexpr const char* runtime_features[] = {
 // When features are added here, there should be a tracking bug assigned to the
 // code owner to flip the default after sufficient testing.
 constexpr const char* disabled_runtime_features[] = {
+    // deprecated features are fatal-by-default
+    "envoy.features.disallow_deprecated_features",
```

Comment:
So runtime guards added here are intended to be flipped true and removed.
Do you want a regular runtime guard rather than a feature with a finite lifetime?

Reply:
I'm not sure if I understand what you mean by regular runtime guard, but I interpreted this as moving it into the runtime features allowlist above and changed the flag name so it could default to true.

Reply:
Re-interpreted this after seeing https://github.com/envoyproxy/envoy/pull/13607/files

Removed from the features list.